managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/cache/RangeEntryCacheImpl.java

```diff
@@ -362,6 +362,9 @@ private AsyncCallbacks.ReadEntriesCallback handlePendingReadsLimits(ReadHandle l
         }
         long estimatedReadSize = (1 + lastEntry - firstEntry)
                 * (estimatedEntrySize + BOOKKEEPER_READ_OVERHEAD_PER_ENTRY);
+        if (estimatedReadSize > pendingReadsLimiter.getMaxReadsInFlightSize()) {
+            pendingReadsLimiter.setMaxReadsInFlightSize(estimatedReadSize);
```

Comment:
Here we are mutating the configuration of a shared object.

We could log an error but in the end the administrator has to change the value

Reply:
address the comment.
Add log here

Reply:
I am sorry, I was not clear.
The point is not about adding only the logging.
But we cannot update the configuration dynamically here.